Sources/CodeEditSourceEditor/Enums/CaptureModifier.swift

```diff
@@ -0,0 +1,133 @@
+/// and passed around, so representing them with a single integer is preferable to a string to save memory.
+///
+public enum CaptureModifier: Int8, CaseIterable, Sendable {
+    case declaration
+    case definition
+    case readonly
+    case `static`
+    case deprecated
+    case abstract
+    case async
+    case modification
+    case documentation
+    case defaultLibrary
+
+    public var stringValue: String {
+        switch self {
+        case .declaration:
+            return "declaration"
+        case .definition:
+            return "definition"
+        case .readonly:
+            return "readonly"
+        case .static:
+            return "static"
+        case .deprecated:
+            return "deprecated"
+        case .abstract:
+            return "abstract"
+        case .async:
+            return "async"
+        case .modification:
+            return "modification"
+        case .documentation:
+            return "documentation"
+        case .defaultLibrary:
+            return "defaultLibrary"
+        }
+    }
+
+    // swiftlint:disable:next cyclomatic_complexity
+    public static func fromString(_ string: String) -> CaptureModifier? {
+        switch string {
+        case "declaration":
+            return .declaration
+        case "definition":
+            return .definition
+        case "readonly":
+            return .readonly
+        case "static`":
+            return .static
+        case "deprecated":
+            return .deprecated
+        case "abstract":
+            return .abstract
+        case "async":
+            return .async
+        case "modification":
+            return .modification
+        case "documentation":
+            return .documentation
+        case "defaultLibrary":
+            return .defaultLibrary
+        default:
+            return nil
+        }
+    }
+}
+
+extension CaptureModifier: CustomDebugStringConvertible {
+    public var debugDescription: String { stringValue }
+}
+
+/// A set of capture modifiers, efficiently represented by a single integer.
+public struct CaptureModifierSet: OptionSet, Equatable, Hashable, Sendable {
+    public var rawValue: UInt
+
+    public init(rawValue: UInt) {
+        self.rawValue = rawValue
+    }
+
+    public static let declaration = CaptureModifierSet(rawValue: 1 << CaptureModifier.declaration.rawValue)
+    public static let definition = CaptureModifierSet(rawValue: 1 << CaptureModifier.definition.rawValue)
+    public static let readonly = CaptureModifierSet(rawValue: 1 << CaptureModifier.readonly.rawValue)
+    public static let `static` = CaptureModifierSet(rawValue: 1 << CaptureModifier.static.rawValue)
+    public static let deprecated = CaptureModifierSet(rawValue: 1 << CaptureModifier.deprecated.rawValue)
+    public static let abstract = CaptureModifierSet(rawValue: 1 << CaptureModifier.abstract.rawValue)
+    public static let async = CaptureModifierSet(rawValue: 1 << CaptureModifier.async.rawValue)
```

Comment:
Does async need to go in backticks?
```suggestion
    public static let `async` = CaptureModifierSet(rawValue: 1 << CaptureModifier.async.rawValue)
```